app/models/work_package.rb

```diff
@@ -460,14 +460,16 @@ def self.available_custom_fields(work_package)
 
   def self.preload_available_custom_fields(work_packages)
     custom_fields = available_custom_fields_from_db(work_packages)
-                    .select('projects.id project_id',
-                            'types.id type_id',
+                    .select('array_agg(projects.id) available_project_ids',
+                            'array_agg(types.id) available_type_ids',
                             'custom_fields.*')
+                    .group('custom_fields.id')
```

Comment:
Very nice 👍. I tried it out and it sets the right criteria for `DISTINCT` from `.available_custom_fields_from_db` when preloading.